src/reflect/scala/reflect/internal/transform/Erasure.scala

```diff
@@ -111,7 +111,7 @@ trait Erasure {
 
     protected def eraseDerivedValueClassRef(tref: TypeRef): Type = erasedValueClassArg(tref)
 
-    def apply(tp: Type): Type = tp match {
+    def apply(tp: Type): Type = tp.dealias match {
```

Comment:
Or maybe not, `RefinedTypes` are not a common case, so it might be better to keep this code direct.